01_BLAST/blast.R

```diff
@@ -0,0 +1,33 @@
+##################################################
+### BLAST sequence alignment implementation
+##################################################
+### Given a specified gene and genome, this code
+### identifies the most likely matching locations
+### in the target genome.
+##################################################
+
+# setup
+if (!requireNamespace("BiocManager", quietly = TRUE)) {
+  install.packages("BiocManager")
+}
+
+BiocManager::install(version="3.19")
+BiocManager::install("rBLAST")
+
+library(rBLAST)
+library(here)
+
+
+# make BLAST database from FASTA file
+makeblastdb(file = here("01_BLAST/data/genome/genome1/ncbi_dataset/data/GCA_000008865.2/GCA_000008865.2_ASM886v2_genomic.fna"),
+            db_name = here("01_BLAST/data/genome/genome1_db/e_coli_db"),
+            dbtype = "nucl")
+
+# set up database object
+db <- blast(db = here("01_BLAST/data/genome/genome1_db/e_coli_db"), type="blastn")
+
+# create sequence object from gene FASTA file
+seq <- readRNAStringSet(here("01_BLAST/data/gene/sequence1.fasta"))
+
+# look for matches
+matches <- predict(db, seq[1])
```

Comment:
Is this seriously the whole thing? That's insanely simple. So the work-flow with rBLAST is to 1) make a database out of the genome, 2) read the gene as an RNA string (I'm slightly confused as to why this is reading as RNA and not DNA, but I'm guessing the FASTA file is in RNA format?), and 3 predict where the RNA belongs on the genome. Is that about right?